src/ucp/rndv/rndv_put.c

```diff
@@ -573,6 +573,28 @@ static void ucp_proto_rndv_put_mtype_frag_completion(uct_completion_t *uct_comp)
     ucp_proto_rndv_ppln_send_frag_complete(req, 1);
 }
 
+static ucs_memory_type_t
+ucp_proto_rndv_put_mtype_frag_mem_type(uint64_t rndv_frag_mem_types,
+                                       ucs_memory_type_t rkey_mem_type)
+{
+    ucs_memory_type_t mem_type;
+
+    if (UCS_BIT_GET(rndv_frag_mem_types, rkey_mem_type)) {
+        return rkey_mem_type;
+    }
+
+    if (UCS_BIT_GET(rndv_frag_mem_types,
+                    UCS_MEMORY_TYPE_HOST)) {
+        return UCS_MEMORY_TYPE_HOST;
+    }
+
+    ucs_for_each_bit(mem_type, rndv_frag_mem_types) {
```

Comment:
WDYT?
```c
if (rndv_frag_mem_types == 0) {
    return UCS_MEMORY_TYPE_UNKNOWN;
} else if (UCS_BIT_GET(rndv_frag_mem_types, rkey_mem_type)) {
    return rkey_mem_type;
 } else if (UCS_BIT_GET(rndv_frag_mem_types,
                     UCS_MEMORY_TYPE_HOST)) {
    return UCS_MEMORY_TYPE_HOST;
 } else {
    return ucs_ffs64(rndv_frag_mem_types);
}
```

Reply:
Looks more concise. Applied the suggestion.